Crash with global variable

Running mypy-0.641 on
```python
from typing import List  # noqa F401


def foo():
    global bar
    bar = []  # type: List[str]
    bar
```
crashed similar to #5081
```
test.py:6: error: INTERNAL ERROR -- please report a bug at https://github.com/python/mypy/issues version: 0.641
Traceback (most recent call last):
  File "/usr/bin/mypy", line 11, in <module>
    load_entry_point('mypy==0.641', 'console_scripts', 'mypy')()
  File "/usr/lib/python3/dist-packages/mypy/__main__.py", line 7, in console_entry
    main(None)
  File "/usr/lib/python3/dist-packages/mypy/main.py", line 92, in main
    res = build.build(sources, options, None, flush_errors, fscache)
  File "/usr/lib/python3/dist-packages/mypy/build.py", line 155, in build
    result = _build(sources, options, alt_lib_path, flush_errors, fscache)
  File "/usr/lib/python3/dist-packages/mypy/build.py", line 204, in _build
    graph = dispatch(sources, manager)
  File "/usr/lib/python3/dist-packages/mypy/build.py", line 2172, in dispatch
    process_graph(graph, manager)
  File "/usr/lib/python3/dist-packages/mypy/build.py", line 2469, in process_graph
    process_stale_scc(graph, scc, manager)
  File "/usr/lib/python3/dist-packages/mypy/build.py", line 2586, in process_stale_scc
    graph[id].semantic_analysis()
  File "/usr/lib/python3/dist-packages/mypy/build.py", line 1710, in semantic_analysis
    self.manager.semantic_analyzer.visit_file(self.tree, self.xpath, self.options, patches)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 302, in visit_file
    self.accept(d)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 3611, in accept
    node.accept(self)
  File "/usr/lib/python3/dist-packages/mypy/nodes.py", line 608, in accept
    return visitor.visit_func_def(self)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 371, in visit_func_def
    self._visit_func_def(defn)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 462, in _visit_func_def
    self.analyze_function(defn)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 721, in analyze_function
    defn.body.accept(self)
  File "/usr/lib/python3/dist-packages/mypy/nodes.py", line 910, in accept
    return visitor.visit_block(self)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 1635, in visit_block
    self.accept(s)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 3611, in accept
    node.accept(self)
  File "/usr/lib/python3/dist-packages/mypy/nodes.py", line 965, in accept
    return visitor.visit_assignment_stmt(self)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 1700, in visit_assignment_stmt
    final_cb=final_cb if s.is_final_def else None)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 1983, in analyze_lvalue
    self.analyze_name_lvalue(lval, add_global, explicit_type, final_cb)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 2048, in analyze_name_lvalue
    self.make_name_lvalue_point_to_existing_def(lval, explicit_type, final_cb)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 2101, in make_name_lvalue_point_to_existing_def
    self.name_already_defined(lval.name, lval, original_def)
  File "/usr/lib/python3/dist-packages/mypy/semanal.py", line 3576, in name_already_defined
    elif node and node.line != -1:
UnboundLocalError: local variable 'node' referenced before assignment
test.py:6: : note: use --pdb to drop into pdb
```

--- a/test-data/unit/check-statements.test
+++ b/test-data/unit/check-statements.test
@@ -1663,3 +1663,19 @@ N = TypedDict('N', {'x': int})
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-full.pyi]
 [out]
+
+[case testGlobalWithoutInitialization]
+# flags: --new-semantic-analyzer
+from typing import List
+
+def foo() -> None:
+    global bar
+    # TODO: Confusing error message
+    bar = []  # type: List[str]  # E: Name 'bar' already defined (possibly by an import)
+    bar  # E: Name 'bar' is not defined
+
+def foo2():
+    global bar2
+    bar2 = []  # type: List[str]
+    bar2
+[builtins fixtures/list.pyi]

